### PR TITLE
Correct RF AI player move

### PR DESCRIPTION
**Pull request by ernie7334066**

### CODE REVIEW

File: src/reinforcementAI.js [hunk 1]
```diff
@@ -84,38 +84,42 @@ export default class ReinforcementAI {
...
     if (SObject.player0dead) {
       return as;
     }
-    var playAgainst = [], guess = [];
+    var playAgainst = [1, 2, 3, 4], guess = [2, 3, 4, 5, 6, 7, 8];
+    var playerID = playAgainst.indexOf(reduxState.currentPlayerId);
+    // Remove players who are dead or protected
+    for (var i = reduxState.players.length-1; i > -1; i--){
+    	if (reduxState.players[i].dead || reduxState.players[i].protected){
+    		playAgainst.splice(i,1);
```
Comment (gges5110) [inline src/reinforcementAI.js]: The default indentation is 2 spaces.

File: src/reinforcementAI.js [hunk 2]
```diff
@@ -84,38 +84,42 @@ export default class ReinforcementAI {
...
+    		playerID = playAgainst.indexOf(reduxState.currentPlayerId);
+    	}    	
+    }
+    var playAgainst2 = playAgainst.slice();
```
Comment (gges5110) [inline src/reinforcementAI.js]: Please give it a better name.

File: src/reinforcementAI.js [hunk 3]
```diff
@@ -84,38 +84,42 @@ export default class ReinforcementAI {
     player0holdingCard1: reduxState.players[0].holdingCards[1],
   }
   */
-  allowedSAVectors(SObject) {
+  allowedSAVectors(SObject,reduxState) {
```
Comment (gges5110) [inline src/reinforcementAI.js]: Try not to pass the whole reduxState around, since it can be huge. Just pass in the necessary data.

Comment (ernie7334066) [reply src/reinforcementAI.js]: Should I include additional data in SObject?

Comment (gges5110) [reply src/reinforcementAI.js]: Yeah you should be able to do something like
```javascript
return {
  player0dead: reduxState.players[1].dead,
  player0lastCardId: player0lastCardId,
  player0holdingCard0: reduxState.players[1].holdingCards[0],
  player0holdingCard1: reduxState.players[1].holdingCards[1],
  // Add you new data here
  currentPlayerId: reduxState.currentPlayerId
}
```

**Pull request closed without merging**